### PR TITLE
V2 Avatar Pie component in Avatar Group

### PR DESCRIPTION
**Pull request by joyeeta26**
The goal of this PR is to add the Avatar Pie component in Fluent v2.

### Validations
Screenshot 2024-10-22 104654
Screenshot 2024-10-22 104934

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] Automated Tests
- [ ] Documentation and demo app examples
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+private fun RenderTwoSlices(
+    avatarSize: Dp, slicedAvatarDimen: Dp, group: Group, size: AvatarSize
+) {
+    Row(
+        modifier = Modifier
+            .requiredSize(avatarSize)
+            .clip(CircleShape)
+    ) {
+        SlicedAvatar(
+            group.members[0],
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:75]: what is he group.member is an empty list. test it once

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    Spacer(
+        modifier = Modifier
+            .background(color = Color.White)
+            .fillMaxWidth()
+            .height(SPACER_SIZE)
+    )
+}
+
+
+private fun getSlicedAvatarSize(size: Dp): SlicedAvatarSize {
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:166]: why is this required. Are these values fluent?

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt [hunk 3]
```diff
@@ -215,6 +208,83 @@ fun Avatar(
     }
 }
 
+@Composable
+fun SlicedAvatar(
+    person: Person,
+    modifier: Modifier = Modifier,
+    width: Dp = 32.dp,
+    avatarToken: AvatarTokens? = null,
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt:216]: pass avatartokens object instead of null

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Utils.kt [hunk 4]
```diff
@@ -152,4 +156,81 @@ fun getAvatarSize(secondaryText: String?, tertiaryText: String?): AvatarSize {
         return AvatarSize.Size40
     }
     return AvatarSize.Size56
+}
+
+
+@Composable
+fun fontTypographyForSlicedAvatar(slicedAvatarSize: SlicedAvatarSize): TextStyle {
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Utils.kt:163]: can you move this avatar tokens instead

Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Utils.kt:163]: but this is not avatar tokens actually
this is not configurable by user but calculated by us based on avatar size
that's why have added it here

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt (hunk 1 above)
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:75]: we have already this check in AvatarGroup to create Avatar Pie only if group.members.size > 1
image

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt (hunk 2 above)
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:166]: removed this

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt [hunk 5]
```diff
@@ -215,6 +208,83 @@ fun Avatar(
...
+    slicedAvatarSize: Dp = 32.dp,
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt:217]: is slicedAvatarSize required. can we use avatarsize instead

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarGroup.kt [hunk 6]
```diff
@@ -81,29 +81,38 @@ fun AvatarGroup(
...
-        if (group.members.size > visibleAvatar || group.members.isEmpty()) {
-            Avatar(
-                group.members.size - visibleAvatar, size = size,
-                enableActivityRings = true, avatarToken = avatarToken
+                noOfVisibleAvatars = visibleAvatar,
+                avatarTokens = avatarToken
             )
+        } else {
+            for (i in 0 until visibleAvatar) {
+                val person = group.members[i]
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarGroup.kt:93]: is ther any chance for group.members be an emptylist?

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+import androidx.compose.ui.unit.dp
+import com.microsoft.fluentui.theme.FluentTheme
+import com.microsoft.fluentui.theme.token.ControlTokens
+import com.microsoft.fluentui.theme.token.controlTokens.AvatarInfo
+import com.microsoft.fluentui.theme.token.controlTokens.AvatarTokens
+import com.microsoft.fluentui.theme.token.controlTokens.AvatarSize
+import com.microsoft.fluentui.theme.token.controlTokens.AvatarType
+import com.microsoft.fluentui.theme.token.controlTokens.SlicedAvatarSize
+
+private val SPACER_SIZE = 2.dp
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:30]: add const

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    val avatarInfo = AvatarInfo(
+        size
+    )
+    val token = avatarTokens
+        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.AvatarControlType] as AvatarTokens
+    val avatarSize = token.avatarSize(avatarInfo)
+
+    if (noOfVisibleAvatars == 1) {
+        BasicText(
+            "Pie should have minimum 2 members visible",
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:45]: BasicText should not be the fallback for an avatarpie. if avatar is only 1 then show only avatar. or if the group is emtpy then return empty composable but a basic text cannot be returned for avatar

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AvatarTokens.kt [hunk 9]
```diff
@@ -150,6 +150,24 @@ enum class AvatarSize {
     Size72
 }
 
+enum class SlicedAvatarSize {
```
Comment (praveen970) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AvatarTokens.kt:153]: why do we need these? where are they used? are these fluent specific?

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Utils.kt [hunk 10]
```diff
@@ -152,4 +158,89 @@ fun getAvatarSize(secondaryText: String?, tertiaryText: String?): AvatarSize {
         return AvatarSize.Size40
     }
     return AvatarSize.Size56
+}
+
+
+@Composable
+fun fontTypographyForSlicedAvatar(slicedAvatarSize: Dp): TextStyle {
```
[the remark this answers is not in the document]
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Utils.kt:165]: no, these are not fluent values

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt (hunk 7 above)
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:30]: it gives error Const 'val' has type 'Dp'. Only primitives and String are allowed

File: fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AvatarTokens.kt (hunk 9 above)
Comment (joyeeta26) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theme/token/controlTokens/AvatarTokens.kt:153]: removed this

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt (hunk 5 above)
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt:217]: slicedAvatarSize would be dependent on the type of slice, for example for half it would be different and for quarter it would be different, hence required

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt (hunk 8 above)
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarPie.kt:45]: fixed

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarGroup.kt (hunk 6 above)
Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/AvatarGroup.kt]: i have added check here that group size should be greater than 0

File: fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt [hunk 11]
```diff
@@ -215,6 +208,83 @@ fun Avatar(
     }
 }
 
+@Composable
```
Comment (praveen970) [inline fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt:211]: is this supposed to be exposed to public, if not can we make it internal or private?

Comment (joyeeta26) [reply fluentui_persona/src/main/java/com/microsoft/fluentui/tokenized/persona/Avatar.kt:211]: done

**Review by praveen970 — APPROVED** [review]

**Review by Dhoot — APPROVED** [review]
